electrum/gui/kivy/uix/dialogs/cpfp_dialog.py

```diff
@@ -0,0 +1,161 @@
+            size_hint_y: None
+            cols: 1
+            spacing: '10dp'
+            BoxLabel:
+                id: total_fee
+                text: _('Total fee')
+                value: ''
+            BoxLabel:
+                id: total_feerate
+                text: _('Total feerate')
+                value: ''
+        Widget:
+            size_hint: 1, 1
+        BoxLayout:
+            orientation: 'horizontal'
+            size_hint: 1, 0.5
+            Button:
+                text: 'Cancel'
+                size_hint: 0.5, None
+                height: '48dp'
+                on_release: root.dismiss()
+            Button:
+                text: 'OK'
+                size_hint: 0.5, None
+                height: '48dp'
+                on_release:
+                    root.dismiss()
+                    root.on_ok()
+''')
+
+class CPFPDialog(FeeSliderDialog, Factory.Popup):
+
+    def __init__(self, app: 'ElectrumWindow', parent_fee, total_size, new_tx, callback):
+        self.parent_fee = parent_fee
+        self.total_size = total_size
+        self.new_tx = new_tx
+        self.max_fee = self.new_tx.output_value()
+        Factory.Popup.__init__(self)
+        FeeSliderDialog.__init__(self, app.electrum_config, self.ids.slider)
+        self.app = app
+        self.callback = callback
+        self.config = app.electrum_config
+        self.ids.total_size.value = ('%d bytes'% self.total_size)
+        self.ids.input_amount.value = self.app.format_amount(self.max_fee) + ' ' + self.app._get_bu()
+        suggested_feerate = self.config.fee_per_kb()
+        self.fee = self.get_child_fee_from_total_feerate(suggested_feerate)
+        self.fee_e = self.app.format_amount_and_units(self.fee)
+        self.ids.fee_for_child.value = self.fee_e
+        self.update_slider()
+        self.update_text()
+
+    def get_child_fee_from_total_feerate(self, fee_per_kb: Optional[int]) -> Optional[int]:
+        if fee_per_kb is None:
+            return None
+        fee = fee_per_kb * self.total_size / 1000 - self.parent_fee
+        fee = round(fee)
+        fee = min(self.max_fee, fee)
+        fee = max(self.total_size, fee)  # pay at least 1 sat/byte for combined size
+        return fee
+
+    def update_text(self):
+        target, tooltip, dyn = self.config.get_fee_target()
+        self.ids.fee_target.text = target
+        fee_per_kb = self.config.fee_per_kb()
+        self.fee = self.get_child_fee_from_total_feerate(fee_per_kb)
+        if fee_per_kb is None:
+            self.ids.fee_for_child.value = "unknown"
+        else:
+            comb_fee = self.fee + self.parent_fee
+            comb_feerate = comb_fee / self.total_size 
+            self.ids.fee_for_child.value = f'{self.fee:.1f} sat'
+            self.ids.output_amount.value = f'{self.max_fee-self.fee:.1f} sat'
+            self.ids.total_fee.value = f'{self.fee+self.parent_fee:.1f} sat'
+            self.ids.total_feerate.value = f'{comb_feerate:.1f} sat/B'
```

Comment:
Please don't format amounts/feerates by hand but use the existing helpers.

```suggestion
            comb_feerate = 1000 * comb_fee / self.total_size
            self.ids.fee_for_child.value = self.app.format_amount_and_units(self.fee)
            self.ids.output_amount.value = self.app.format_amount_and_units(self.max_fee-self.fee)
            self.ids.total_fee.value = self.app.format_amount_and_units(self.fee+self.parent_fee)
            self.ids.total_feerate.value = self.app.format_fee_rate(comb_feerate)
```
Note that in order to be able to reference `self.app` here; in `__init__`, you might have to move its assignment (`self.app = app`) to a few lines earlier.